### PR TITLE
Matrix OSRM Compatible Response

### PR DESCRIPTION
**Pull request by kdiluca**
1) This is waiting to be switched out with the waypoints serialization work
2) Need assistance as to why the test fails (currently commented out)

Sample local test url:  http://localhost:8002/sources_to_targets?json={"sources":[{"lat":52.517037,"lon":13.388860},{"lat":52.524525,"lon":13.416109}],"targets":[{"lat":52.517037,"lon":13.388860},{"lat":52.529407,"lon":13.397634},{"lat":52.524525,"lon":13.416109}],"costing":"auto"}&format=osrm

### CODE REVIEW

File: src/tyr/matrix_serializer.cc [hunk 1]
```diff
@@ -12,69 +12,141 @@ using namespace valhalla::thor;
 
 namespace {
```
Comment (kevinkreiser) [inline src/tyr/matrix_serializer.cc]: no need to even have the anonymous namespace

**Comment by kevinkreiser** [comment]
just a quick FYI, the waypoints branch has `odin::Location` pushed all the way through to the matrix stuff so you can work on that branch to do the waypoints serialization. the reason that branch is not building currently is because of route having so many uses of `baldr::PathLocation`. so anyway i guess im saying writing the waypoints serialization code isnt blocked but the build is broken because there is more work to do to remove `baldr::PathLocation` use from thor.

**Comment by kdiluca** [comment]
Ok, I'll go ahead and get your branch working along with using the updated waypoint serializer.  Thanks!

**Comment by kevinkreiser** [comment]
@kdiluca Yeah its quite a bit of work yet unfortunately... But basically replace all instances of `baldr::PathLocation` in thor with `odin::Location`. Its pretty mechanical except for one thing:

`odin::Location` has path edges. In trip path builder we need to know which one was used for the route at the origin and destination. There is a little loop in there which matches the ids. When we do that. we need to delete all the other path edges so that there is only one left for a given origin or destination. pbf objects are annoying in this way because even the mutable repeated fields dont let you remove elements from the middle (only at the back). so you'll have to find the matching one, swap it with the begging element and then delete until there is only one left in the mutable repeated path edges. i hope this makes sense but if not feel please send me questions.

**Comment by kdiluca** [comment]
The Matrix work has been merged into Valhalla master via the Waypoints branch so closing this PR.

**Pull request closed without merging**